owltorrent-network/src/main/java/edu/rice/owltorrent/network/messages/PieceActionMessage.java

```diff
@@ -24,13 +24,21 @@
   private final int begin;
   private final int length;
 
-  public PieceActionMessage(MessageType type, int index, int begin, int length) {
+  private PieceActionMessage(MessageType type, int index, int begin, int length) {
```

Comment:
Why aren't we just using two normal Message classes with constructors for this? Confusing and means that it doesn't get to inherit and use the same abstractions and methods

Reply:
Sorry can you elaborate on what you mean? This class is basically the abstraction of Request and Cancel messages. They share the exact same format except the message type.

Reply:
Ah my bad I thought it didn't extend PieceMessage for some reason